# HG changeset patch
# User Rik <rik@octave.org>
# Date 1553571075 25200
#      Mon Mar 25 20:31:15 2019 -0700
# Node ID eebf4b67f012b96f2fa5eae5a2c9ec6ec35eb97d
# Parent  de826e69a5ea56a4f9e7790e43472f30386ebec3
Silence warning on Mac platforms about deprecated OpenGL API (bug #55991).

* oct-conf-post.in.h: If on Mac, and building OpenGL plotting, define
GL_SILENCE_DEPRECATION so that hundreds of warnings aren't printed.

diff --git a/oct-conf-post.in.h b/oct-conf-post.in.h
--- a/oct-conf-post.in.h
+++ b/oct-conf-post.in.h
@@ -119,16 +119,21 @@ typedef unsigned long ino_t;
    /* deprecated function names (FIXME: ???) */
 #  pragma warning (disable: 4661)
 #endif
 
 #if defined (__APPLE__) && defined (__MACH__)
 #  define OCTAVE_USE_OS_X_API 1
 #endif
 
+/* Silence deprecated API warning from Apple OS > 10.14 */
+#if defined (__APPLE__) && defined (__MACH__) && defined (HAVE_OPENGL)
+#  define GL_SILENCE_DEPRECATION 1
+#endif
+
 /* Define to 1 if we expect to have <windows.h>, Sleep, etc. */
 #if defined (__WIN32__) && ! defined (__CYGWIN__)
 #  define OCTAVE_USE_WINDOWS_API 1
 #endif
 
 #if defined (OCTAVE_USE_WINDOWS_API)
 #  define OCTAVE_HAVE_WINDOWS_FILESYSTEM 1
 #elif defined (__CYGWIN__)
